pact/hft.pact

```diff
@@ -165,194 +134,194 @@
+        { "guard" := g }
+        (credit id receiver g amount))
+    )
+  )
 
-    (defun truncate:decimal (token:string amount:decimal)
-      (floor amount (precision token))
+  (defun transfer-create:string
+    ( id:string
+      sender:string
+      receiver:string
+      receiver-guard:guard
+      amount:decimal
     )
+    (enforce (!= sender receiver)
+      "sender cannot be the receiver of a transfer")
+    (enforce-valid-transfer sender receiver (precision id) amount)
+
+    (with-capability (TRANSFER id sender receiver amount)
+      (debit id sender amount)
+      (credit id receiver receiver-guard amount))
+  )
 
-    (defun get-balance:decimal (token:string account:string)
-      (at 'balance (read ledger (key token account)))
-      )
-
-    (defun details
-      ( token:string account:string )
-      (read ledger (key token account))
-      )
-
-    (defun rotate:string (token:string account:string new-guard:guard)
-      (with-capability (ROTATE token account)
-        (with-read ledger (key token account)
-          { "guard" := old-guard }
-
-          (enforce-guard old-guard)
-          (update ledger (key token account)
-            { "guard" : new-guard }))))
-
-    (defun transfer:string
-      ( token:string
-        sender:string
-        receiver:string
-        amount:decimal
-      )
-      (enforce (!= sender receiver)
-        "sender cannot be the receiver of a transfer")
-      (enforce-valid-transfer sender receiver (precision token) amount)
-
-      (with-capability (TRANSFER token sender receiver amount)
-        (debit token sender amount)
-        (with-read ledger (key token receiver)
-          { "guard" := g }
-          (credit token receiver g amount))
-      )
+  (defun mint:string
+    ( id:string
+      account:string
+      guard:guard
+      amount:decimal
     )
+    (with-capability (MINT id account amount)
+      (with-capability (CREDIT id account)
+        (credit id account guard amount)
+        (update-supply id amount)))
+  )
 
-    (defun transfer-create:string
-      ( token:string
-        sender:string
-        receiver:string
-        receiver-guard:guard
-        amount:decimal
-      )
-      (enforce (!= sender receiver)
-        "sender cannot be the receiver of a transfer")
-      (enforce-valid-transfer sender receiver (precision token) amount)
-
-      (with-capability (TRANSFER token sender receiver amount)
-        (debit token sender amount)
-        (credit token receiver receiver-guard amount))
-      )
-
-    (defun mint:string
-      ( token:string
-        account:string
-        guard:guard
-        amount:decimal
-      )
-      (with-capability (MINT token account amount)
-        (with-capability (CREDIT token account)
-          (credit token account guard amount)))
+  (defun burn:string
+    ( id:string
+      account:string
+      amount:decimal
     )
+    (with-capability (BURN id account amount)
+      (with-capability (DEBIT id account)
```

Comment:
Same with above comment in `mint`. Unnecessary because DEBIT is composed in BURN.